playlist_souffle/use_cases/souffle_playlist.py

```diff
@@ -0,0 +1,36 @@
+"""Module for the souffle_playlist use case."""
+
+from datetime import datetime as dt
+from playlist_souffle.definitions.playlist import Playlist
+import playlist_souffle.entities.souffler as souffler_entity
+import playlist_souffle.entities.track as track_entity
+
+def souffle_playlist(spotify, playlist_uri, user_id, shuffle_by):
+    """Souffle a playlist."""
+
+    original_playlist = Playlist(
+        owner=None,
+        name=spotify.fetch_playlist_name(playlist_uri),
+        tracks=spotify.fetch_playlist_tracks(playlist_uri),
+        description=None
+    )
+
+    collection_id_by_track = {track: track_entity.extract_track_collection_id(track, shuffle_by)
```

Comment:
perhaps there's a `track_entity.get_collection_id_by_track(original_playlist)` ?

Reply:
Again i usually dont like creating functions that just encapsulate a dict / list comprehension.

I generally think dict/list comprehensions are super readable once youre in the python mindset. Also then i'd end up testing a function that just does a comprehension

Reply:
I really like list comprehensions. if they were in an entity, I wouldn't be encouraging you to move them. But I like the use case having as little logic as possible. it's just matters what your priority scheme is here. either way works just fine.

Reply:
Btw, in your comment on the use case you wrote:

> And actually a list interpretation makes it clearer: for each track, fetch related tracks from spotify service

Am i misunderstanding (these are addressing different lines) or are you wide & contain contradictions?

Reply:
I think I was tired and containing contradictions. :)

I think I found the list mindset easier to explain in human language what was happening, but also wanted business logic in the entity. So yeah, there's a contradiction there.